falcon_kit/stats_preassembly.py

```diff
@@ -115,7 +116,25 @@ def read_lens_from_db(db_fn):
     """
     return list(sorted(get_db_readlengths(db_fn)))
 
-def stats_dict(stats_raw_reads, stats_seed_reads, stats_corrected_reads, genome_length, length_cutoff):
+def metric_fragmentation(preads_dir):
+    # https://jira.pacificbiosciences.com/browse/SAT-105
+    bash = r"""
+sed -nr 's;>prolog/([0-9]*)[0-9]/.*;\1;p' %s/*.fasta | sort | uniq -c | awk '{print $1}' | sort | uniq -c
```

Comment:
That sed command I gave you is pretty inefficient. Why not just do this in pure python with a hashtable?